qiskit/transpiler/passmanager.py

```diff
@@ -24,14 +25,15 @@
 from .exceptions import TranspilerError
 from .runningpassmanager import RunningPassManager, FlowController
 
-
+# TODO: why not
+#  _CircuitsT = Union[List[QuantumCircuit], QuantumCircuit]
```

Comment:
There are errors like 
```

Reply:
I guess it is because the `bound`, which allows typing against the subsets.

Reply:
I don't think it works:
```
qiskit/transpiler/passmanager.py:233: error: Incompatible return value type (got "QuantumCircuit", expected "_CircuitsT")  [return-value]
qiskit/transpiler/passmanager.py:235: error: Incompatible return value type (got "List[QuantumCircuit]", expected "_CircuitsT")  [return-value]
qiskit/transpiler/passmanager.py:235: error: Argument 1 to "_run_single_circuit" of "PassManager" has incompatible type "Union[QuantumCircuit, CircuitInstruction]"; expected "QuantumCircuit"  [arg-type]
qiskit/transpiler/passmanager.py:236: error: Incompatible return value type (got "List[QuantumCircuit]", expected "_CircuitsT")  [return-value]
qiskit/transpiler/passmanager.py:236: error: Argument 1 to "_run_several_circuits" of "PassManager" has incompatible type "_CircuitsT"; expected "List[QuantumCircuit]"  [arg-type]
```
These five errors are fixed if I switch current and proposed definitions and no new errors appear.

Reply:
So... what's the `TODO`?

Reply:
If there is some reason why the current form is used, to fix the type errors. Otherwise, replace with proposed definition